From 17cd68ff29a535f0795cbc632925e6ad815c5cb7 Mon Sep 17 00:00:00 2001
From: Sean Mooney <work@seanmooney.info>
Date: Tue, 12 Jan 2021 20:42:08 +0000
Subject: [PATCH 1/7] add constants for vnic type vdpa

This change extend the nova.network.model module to define the vdpa
vnic type added in I01083fb8192b2ef00934af619576889b881c1249

Depends-On: https://review.opendev.org/c/openstack/neutron-lib/+/760043
Partial-Implements: blueprint libvirt-vdpa-support
---
 nova/network/model.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/nova/network/model.py b/nova/network/model.py
index 416317b18b..f0f04a9f6e 100644
--- a/nova/network/model.py
+++ b/nova/network/model.py
@@ -104,14 +104,16 @@ VNIC_TYPE_MACVTAP = 'macvtap'
 VNIC_TYPE_DIRECT_PHYSICAL = 'direct-physical'
 VNIC_TYPE_BAREMETAL = 'baremetal'
 VNIC_TYPE_VIRTIO_FORWARDER = 'virtio-forwarder'
+VNIC_TYPE_VDPA = 'vdpa'
 
 # Define list of ports which needs pci request.
 # Note: The macvtap port needs a PCI request as it is a tap interface
 # with VF as the lower physical interface.
 # Note: Currently, VNIC_TYPE_VIRTIO_FORWARDER assumes a 1:1
 # relationship with a VF. This is expected to change in the future.
-VNIC_TYPES_SRIOV = (VNIC_TYPE_DIRECT, VNIC_TYPE_MACVTAP,
-                    VNIC_TYPE_DIRECT_PHYSICAL, VNIC_TYPE_VIRTIO_FORWARDER)
+VNIC_TYPES_SRIOV = (
+    VNIC_TYPE_DIRECT, VNIC_TYPE_MACVTAP, VNIC_TYPE_DIRECT_PHYSICAL,
+    VNIC_TYPE_VIRTIO_FORWARDER, VNIC_TYPE_VDPA)
 
 # Define list of ports which are passthrough to the guest
 # and need a special treatment on snapshot and suspend/resume
-- 
2.17.1

